rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+ensuring that their privacy settings remain aligned with their needs as they change.
+
+This approach not only empowers users but also facilitates greater adoption of Dapps. Users who are skeptical about privacy concerns can initially engage with transparent 
+transactions and gradually shift towards more private executions as they gain confidence in the system, and vice versa for users who start with privacy but later find transparency 
+beneficial for certain transactions. In this way, selective privacy bridges the gap between privacy and transparency, allowing for an optimal balance to emerge from the community’s 
+collective preferences.
+
+To liken this to open-source projects: in traditional systems, developers fix privacy rules much like immutable code—users must comply with these fixed settings. 
+In contrast, with selective privacy, the rules are malleable and shaped by the users’ preferences, enabling the community to find the ideal balance between privacy and efficiency over time.
+
+NSSA is distinct from traditional zero-knowledge (ZK) roll-ups in several key ways. One of the unique features of NSS is its **public execution type**, which does not 
+require Zero-Knowledge Proofs or a zkVM. This provides a significant advantage in terms of scalability and efficiency, as users can choose public executions for 
+transactions that do not require enhanced privacy, avoiding the overhead associated with ZKP generation and verification.
+
+Moreover, NSSA introduces two additional execution types—**shielded and deshielded**—which further distinguish it from traditional privacy-preserving roll-ups. 
+These execution types allow for more nuanced control over privacy, giving users the ability to shield certain aspects of a transaction while deshielding others. 
+This flexibility sets NSSA apart as a more adaptable and user-centric platform, catering to a wide range of privacy needs without imposing a one-size-fits-all solution.
+
+By combining selective privacy with a flexible execution model, Nescience State Separation architecture offers a more robust and adaptable framework for decentralized applications, 
+ensuring that users maintain control over their privacy while benefiting from the security and efficiency of blockchain technology.
+
+### How Nescience State Separation Architecture (NSSA) Can Be Used
+
+The **Nescience State Separation Architecture offers a flexible, privacy-preserving add-on that can be applied to existing decentralized applications (Dapps). 
+One of the emerging trends in the blockchain space is that each Dapp is expected to have its own rollup for efficiency, and it is estimated that Ethereum could see 
+the deployment of millions of rollups in the near future. A key question arises: how many of these rollups will incorporate privacy? For Dapp developers who want to offer flexible, 
```

Comment:
Is "millions" a bit sensationalist? I know there will be a lot, but this seems a crazy estimate (for the "near future" anyway)?